### PR TITLE
Release 0.8.0

### PR DESCRIPTION
**Pull request by blootsvoets**
Changes since version 0.7:
- Moved DoubleArrayCollector to AggregateListCollector
- Moved DoubleCollector to NumericAggregateCollector
- Fixed MockProducer closing sequence to avoid deadlock
- Added configurable ObservationKey to CsvGenerator

### CODE REVIEW

File: src/main/java/org/radarcns/stream/collector/AggregateListCollector.java [hunk 1]
```diff
@@ -16,26 +16,51 @@
...
+    /** Array collector without schema. Double entries can be added, but entire records cannot. */
+    public AggregateListCollector(String[] fieldNames) {
+        this(fieldNames, null);
+    }
+
+    /** Array collector with schema. Double entries or entire records can be added. */
+    public AggregateListCollector(String[] fieldNames, Schema schema) {
+        collectors = new NumericAggregateCollector[fieldNames.length];
+        for (int i = 0; i < collectors.length; i++) {
+            collectors[i] = new NumericAggregateCollector(fieldNames[i], schema);
```
Comment (nivemaham) [inline src/main/java/org/radarcns/stream/collector/AggregateListCollector.java]: Do we assume that all the fields would have the same schema?

Comment (blootsvoets) [reply src/main/java/org/radarcns/stream/collector/AggregateListCollector.java]: Yes, I'll add that to the documentation.

Comment (blootsvoets) [reply src/main/java/org/radarcns/stream/collector/AggregateListCollector.java]: B.t.w., it's currently used with for example gyroscopes or accelerometers that have multiple axes.

File: src/main/java/org/radarcns/stream/collector/NumericAggregateCollector.java [hunk 2]
```diff
@@ -0,0 +1,203 @@
...
+                    && fieldType != Type.LONG) {
+                throw new IllegalArgumentException("Field " + fieldName + " is not a number type.");
+            }
+        }
+    }
+
+    @Override
+    public NumericAggregateCollector add(SpecificRecord record) {
+        if (pos == -1) {
+            throw new IllegalStateException(
```
Comment (nivemaham) [inline src/main/java/org/radarcns/stream/collector/NumericAggregateCollector.java]: In this case, should we have the `public NumericAggregateCollector(String fieldName)` constructor? I do not see another way of setting a value for `pos`.

Comment (blootsvoets) [reply src/main/java/org/radarcns/stream/collector/NumericAggregateCollector.java]: In that case it is still possible to create derived aggregates. So for Empatica E4, we can derive heart rate from inter-beat-interval, by just adding a value as a double and giving it the "heartRate" field name. However, then the `add(SpecificRecord)` method may not be used.

**Comment by blootsvoets** [comment]
@nivemaham Like for the NumericAggregateCollector, adding the constructor with just `fieldNames` parameter makes it possible to compute derived values as well as single-record field values.

**Pull request merged**